snowdrop-util/src/Snowdrop/Util/Helpers.hs

```diff
@@ -50,6 +51,31 @@ class VerifySigned sigScheme a where
     verifySignature :: PublicKey sigScheme -> a -> Signature sigScheme a -> Bool
     verifySignature pk msg = verifySigned . Signed msg pk
 
+    -- | Verify signature of the supplied data which is assumed to be made by a given public key
+    verifySignedNoMessage :: SignedNoMessage sigScheme a -> a -> Bool
+    verifySignedNoMessage (SignedNoMessage pk sig) msg = verifySignature pk msg sig
+
+-- | Helper data type to hold data along with its signature
+-- and a public key corresponding to the signature.
+data SignedNoMessage sigScheme msg = SignedNoMessage
```

Comment:
Not used now, to be used later for utxo witnesses

Reply:
>NoMessage

Are you going to introduce no-types and antimatter?

jokes aside, maybe it makes sense to add `PublicKey` to `Signed` then it will be able to verify it, what make it more completed and also stop from introducing new vague datatypes?